viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -745,7 +761,9 @@ void CartoFacade::Stop() {
         throw VIAM_CARTO_NOT_IN_STARTED_STATE;
     }
     state = CartoFacadeState::IO_INITIALIZED;
-    StopSaveInternalState();
+    if (!config.cloud_story_enabled) {
+        StopSaveInternalState();
+    }
```

Comment:
Moving cloud story enabled logic from `viam_cartographer.go` to `carto_facade.cc` so that proper state changes can be set and checked still